gems/aws-sdk-core/spec/aws-sdk-core/config-file-sso-session-parser-tests.json

```diff
@@ -0,0 +1,720 @@
+            }
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Sub-property definitions cannot have an empty name.",
+      "input": {
+        "configFile": "[sso-session foo]\ns3 =\n = value"
+      },
+      "output": {
+        "errorContaining": "Property did not have a name in sub-property"
+      }
+    },
+
+    {
+      "name": "Sub-property definitions cannot have an invalid name.",
+      "input": {
+        "configFile": "[sso-session foo]\ns3 =\n in valid = value"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "s3": {}
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Sub-properties can have blank lines that are ignored",
+      "input": {
+        "configFile": "[sso-session foo]\ns3 =\n name = value\n\t \n name2 = value2"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {
+            "s3": {
+              "name": "value",
+              "name2": "value2"
+            }
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Config profiles without prefix are ignored.",
+      "input": {
+        "configFile": "[foo]\nname = value"
+      },
+      "output": {
+        "ssoSessions": {}
+      }
+    },
+
+    {
+      "name": "Comment characters adjacent to sso-session decls",
+      "input": {
+        "configFile": "[sso-session foo]; semicolon\n[sso-session bar]# pound"
+      },
+      "output": {
+        "ssoSessions": {
+          "foo": {},
+          "bar": {}
+        }
+      }
+    },
+
+    {
+      "name": "Invalid continuation",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\n[sso-session foo]\n -continued"
+      },
+      "output": {
+        "errorContaining": "Expected a property definition, found continuation"
+      }
+    },
+
+    {
+      "name": "profile name with no space after `sso-session` is invalid",
+      "input": {
+        "configFile": "[sso-sessionfoo]\nname = value\n[sso-session bar]"
+      },
+      "output": {
+        "ssoSessions": {
+          "bar": {}
+        }
+      }
+    },
+
+    {
+      "name": "sso-session name with extra whitespace",
+      "input": {
+        "configFile": "[   sso-session foo    ]\nname = value\n[sso-session bar]"
+      },
+      "output": {
+        "ssoSessions": {
+          "bar": {},
+          "foo": {
+            "name": "value"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "properties from an invalid sso-session name are ignored",
+      "input": {
+        "configFile": "[sso-session foo]\nname = value\n[sso-session in valid]\nx = 1\n[sso-session bar]\nname = value2"
+      },
+      "output": {
+        "ssoSessions": {
+          "bar": {
+            "name": "value2"
+          },
+          "foo": {
+            "name": "value"
+          }
+        }
+      }
+    },
+
+    {
+      "name": "Duplicate properties in duplicate profiles use the last one defined (case insensitive).",
```

Comment:
Should maybe add specific test case for profile foo + sso-session foo don't share properties or something like that.